deploy/api-gateway/api-deployment.tf

```diff
@@ -9,10 +9,35 @@ resource "aws_api_gateway_deployment" "deployment" {
     "aws_api_gateway_integration.token_integration",
     "aws_api_gateway_integration.account_exchanges_integration"]
 
-  rest_api_id = "${aws_api_gateway_rest_api.api.id}"
-  stage_name = "${var.env}"
+  rest_api_id = aws_api_gateway_rest_api.api.id
+  stage_name = var.env
   variables = {
     deployed_at = timestamp()
   }
+  lifecycle {
+    create_before_destroy = true
```

Comment:
What does this do ?

Reply:
I encountered an issue for redeploying the API after having added the mapping with the domain name. Following this : https://github.com/hashicorp/terraform/issues/10674 I added this snippet and it works. In the issue they provide an other, more detailed solution where they create explicit stages and set dependencies in terraform. If you can have a look and telle me which one seems better it would be great. I will have a try at the other solution to see if it works.

Reply:
Basically, what this one does is : if you have to destroy this deployment (which it does as we set a new variable to enforce it at every deployment), create the new one first and then destroy the old one. This way, the stage is attached to the new deployment and then the old one is deleted.

I find this more elegant than setting an explicit stage (we have one embeded in the deployment  resource) and explicit `depends_on` blocks in several resources. Anyway, this is advanced Terraform and API Gateway specific behavior, so it will be difficult to abstract these strange behaviors.